src/lib/Libpython/pbs_python_svr_internal.c

```diff
@@ -8994,9 +8994,9 @@ pbsv1mod_meth_iter_nextfunc(PyObject *self, PyObject *args, PyObject *kwds)
 					/* skip jobs according to filters requested for the iterator */
 					job *njob = (job *) iter_entry->data;
 					while (njob != NULL &&
-						((ignore_fin && njob->ji_qs.ji_state == JOB_STATE_FINISHED) ||
+						((ignore_fin && check_job_state(njob, JOB_STATE_LTR_FINISHED)) ||
```

Comment:
I've removed ji_qs. ji_state. So, we don't store numeric states anymore, we only store letter states directly in the job_state attribute, which is of type char. So, all of the comparisons needed to be converted to check for the letter states instead of numeric.

Reply:
oh ok. In that case, how are you handling upgrade scenarios? I think both mom and server upgrades will be affected.